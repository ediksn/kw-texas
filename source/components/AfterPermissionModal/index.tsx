import React from 'react'
import { useTranslation } from 'react-i18next'
import { Linking, Platform, Text, View } from 'react-native'
import Modal from 'react-native-modal'
import Button from '../Button'
import { styles } from './styles'

interface Props {
  isVisible: boolean
  biometryType: string
  handleCloseModal: () => void
}
const AfterPermissionModal = ({ isVisible, biometryType, handleCloseModal }: Props) => {
  const { t } = useTranslation()

  const handleGoToSettings = () => {
    handleCloseModal()
    if (Platform.OS === 'ios') {
      Linking.openURL('app-settings://permissions/myapp')
    } else {
      Linking.openSettings()
    }
  }
  return (
    <Modal isVisible={isVisible} onBackdropPress={() => null}>
      <View style={styles.container}>
        <Text style={styles.titleText}>{t(`components_AfterPermissionModal_TurnOn${biometryType}`)}</Text>
        <Text style={styles.bodyText}>{t(`components_AfterPermissionModal_Opportunity${biometryType}`)}</Text>
        <View style={styles.buttonsContainer}>
          <Button
            viewStyle={styles.button}
            message={t('components_AfterPermissionModal_GoToSettings')}
            type='TEXT'
            onPress={handleGoToSettings}
          />
          <Button viewStyle={styles.button} message={t('general_text_Ok')} onPress={handleCloseModal} />
        </View>
      </View>
    </Modal>
  )
}

export default AfterPermissionModal
