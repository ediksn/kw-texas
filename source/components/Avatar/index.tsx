import React, { useState } from 'react'
import { Image, ImageStyle, Text, TextStyle, View, ViewStyle } from 'react-native'
import DefaultAvatar from 'assets/images/default-avatar.png'
import { useSelector } from 'react-redux'
import { styles } from './styles'
import { useGetInitials } from '~/hooks'
import { ProfileInterface } from '~/interfaces/usrInterface'
import { RootState } from '~/store'

interface Props {
  uri?: any
  testID?: string
  avatarStyle?: ImageStyle
  avatarDefaultStyle?: ImageStyle
  initialsViewStyle?: ViewStyle
  initialsStyle?: TextStyle
  forceInitials?: boolean
}

const Avatar = ({
  uri,
  testID,
  avatarStyle,
  avatarDefaultStyle,
  initialsViewStyle,
  initialsStyle,
  forceInitials = false
}: Props) => {
  const profiles: ProfileInterface[] = useSelector((state: RootState) => state.usrProfile.profiles)
  const activeAccount: number = useSelector((state: RootState) => state.usrProfile.activeAccount)
  const currentProfile = (profiles && !!profiles.length && profiles[activeAccount]) || null
  const [imageError, setImageError] = useState(false)
  const uriOk = typeof uri !== 'undefined' && uri !== ''
  const imageOk = !imageError && uriOk

  return imageOk ? (
    <Image
      testID={testID}
      resizeMode='cover'
      source={{ uri }}
      style={[styles.avatar, avatarStyle]}
      onError={() => !imageError && setImageError(true)}
    />
  ) : !imageOk && forceInitials ? (
    <View style={[styles.initialsView, initialsViewStyle]}>
      <Text style={[styles.initials, initialsStyle]}>{useGetInitials(currentProfile?.name)}</Text>
    </View>
  ) : (
    <Image
      testID={testID}
      resizeMode='cover'
      source={DefaultAvatar}
      style={[styles.avatarDefault, avatarDefaultStyle]}
    />
  )
}

export default Avatar
