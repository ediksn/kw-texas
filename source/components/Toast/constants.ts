import { verticalScale } from 'react-native-size-matters'

export const TOAST_HEIGHT = verticalScale(48)
export const TOAST_TYPE = {
  ERROR: 'ERROR',
  SUCCESS: 'SUCCESS'
}
