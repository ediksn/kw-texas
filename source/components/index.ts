/* eslint-disable import/no-cycle */
export { default as Avatar } from './Avatar'
export { default as Button } from './Button'
export { default as CheckBox } from './CheckBox'
export { default as Header } from './Header'
export { default as Input } from './Input'
export { default as Spinner } from './Spinner'
export { default as Video } from './NativeVideo'
export { default as VideoCard } from './VideoCard'
export { default as VideoList } from './VideoList'
export { default as HeaderTitleBox } from './HeaderTitleBox'
export { default as Post } from './Post'
export { default as Icon } from './Icon'
export { default as Dropdown } from './Dropdown'
export { default as Toast } from './Toast'
export { default as EventCard } from './EventCard'
export { default as EmptyList } from './EmptyList'
export { default as EventsCarrousel } from './Carrousel'
export { default as PostInfo } from './PostInfo'
export { default as PostMedia } from './PostMedia'
export { default as PostLinksForm } from './PostLinksForm'
export { default as TagsPeopleHeader } from './TagsPeopleHeader'
export { default as Tags } from './Tags'
