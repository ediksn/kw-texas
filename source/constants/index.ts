export { FORM } from './form'
export { NAVIGATION } from './navigation'
export { theme } from './theme'
export { ACTION } from './action'
export * from './statis'
export * from './testIds'
