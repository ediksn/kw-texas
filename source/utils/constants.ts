export const MAX_CHARACTERS_NEW_POST: number = 500
export const POST_PREVIEW_MEDIA_LENGTH: number = 3
export const GROUP_PREVIEW_MEDIA_LENGTH: number = 10
