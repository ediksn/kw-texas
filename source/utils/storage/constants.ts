export default {
  LOGIN: {
    USER: 'LOGIN.USER',
    SESSION: 'LOGIN.SESSION'
  },
  BIOMETRIC: {
    PERMISSION: 'BIOMETRIC.PERMISSION',
    CREDENTIALS: 'BIOMETRIC.CREDENTIALS'
  }
}
