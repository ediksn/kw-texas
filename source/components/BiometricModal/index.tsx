/** @format */

import React from 'react'
import { useTranslation } from 'react-i18next'
import { Text, TouchableOpacity, View } from 'react-native'
import ReactNativeBiometrics from 'react-native-biometrics'

import { Icon } from '~/components'
import { theme } from '~/constants'

import { styles } from './styles'

interface Props {
  onAuth: any
  biometryType: string
  setBiometricPosition: React.Dispatch<React.SetStateAction<number>>
}

const BiometricModal = ({ onAuth, biometryType, setBiometricPosition }: Props) => {
  const { t } = useTranslation()

  const texts: { [index: string]: string } = {
    [ReactNativeBiometrics.FaceID]: t('components_BiometricModal_FaceID'),
    [ReactNativeBiometrics.Biometrics]: t('components_BiometricModal_Fingerprint'),
    [ReactNativeBiometrics.TouchID]: t('components_BiometricModal_Fingerprint')
  }
  const icons: { [index: string]: string } = {
    [ReactNativeBiometrics.FaceID]: 'faceid-icon',
    [ReactNativeBiometrics.TouchID]: 'fingerprint-icon',
    [ReactNativeBiometrics.Biometrics]: 'fingerprint-icon'
  }

  return (
    <View
      style={styles.container}
      onLayout={event => {
        setBiometricPosition(event.nativeEvent.layout.y)
      }}
    >
      <Text style={styles.text}>{texts[biometryType]}</Text>
      <TouchableOpacity onPress={onAuth}>
        <Icon size={40} name={icons[biometryType]} color={theme.activeColor} />
      </TouchableOpacity>
    </View>
  )
}

export default BiometricModal
