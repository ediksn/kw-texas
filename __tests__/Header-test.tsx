import React from 'react'
import renderer from 'react-test-renderer'
import { Header } from '~/components'
import { theme } from '~/constants'

test('renders correctly', () => {
  const tree = renderer.create(<Header title='Home' backgroundColor={theme.backgroundColor} />).toJSON()
  expect(tree).toMatchSnapshot()
})
