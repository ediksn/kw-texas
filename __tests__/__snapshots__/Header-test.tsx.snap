// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomColor": "rgb(200,200,200)",
        "borderBottomWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "left": 0,
        "maxWidth": "100%",
        "paddingVertical": "2%",
        "position": "absolute",
        "top": 0,
        "width": "100%",
      },
      Object {
        "backgroundColor": "#FFFFFF",
        "paddingTop": undefined,
      },
    ]
  }
>
  <View
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": "auto",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 17,
          "fontWeight": "bold",
        }
      }
    >
      Home
    </Text>
  </View>
</View>
`;
